pandas/tests/indexing/test_coercion.py

```diff
@@ -331,7 +331,7 @@ def test_setitem_index_float64(self, val, exp_dtype, request):
         if exp_dtype is IndexError:
             # float + int -> int
             temp = obj.copy()
-            with pytest.raises(exp_dtype):
+            with tm.external_error_raised(exp_dtype):
```

Comment:
what is the error?

Reply:
`IndexError: index 5 is out of bounds for axis 0 with size 4`

Reply:
i think its worth matching this

Reply:
done